[FIX] table_from_frame: replace nan with String.Unknown for string variable

PrimozGodec opened the pull request:
##### Issue
<!-- E.g. Fixes #1, Closes #2, Resolves #3, etc. -->
<!-- Or a short description, if the issue does not exist. -->
1. When transforming pandas data frame to table string columns that contain nan values will 
keep them after the transformation but Orange uses an empty string for the unknown value in the StringVarable. 

   For example:
      df = pd.DataFrame(
           [["a", "b"], ["c", "d"], ["e", "f"], [np.nan, np.nan]],
       )

   will be transformed to the table with two string variables and nan values will be kept even `String.Unknow = ""`

2. When a column is recognized to be a string and has an object type, it still can contain some values that are not strings. Cast column to string.

##### Description of changes
Changed that nan values are transformed to `String.Unknown` for columns that will be transformed to the string variable and values are transformed to strings.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

markotoplak commented:
Could you better explain what is the reasoning here?

Does not Orange use the empty string OR nan? I'd lean towards using nan wherever possible. And this goes further from what I'd like to see. :)

PrimozGodec commented:
> Could you better explain what is the reasoning here?

I added a better explanation of the issue.

> Does not Orange use the empty string OR nan? I'd lean towards using nan wherever possible. And this goes further from what I'd like to see. :)

I am not sure if nan is supported by all widgets (at least it does not work in Orange3-text addon). I can start working on supporting it there. I changed my implementation that it uses StringVariable.Unknow instead of hardcoded "" such that it will still work when we decide to use nan instead of "".
